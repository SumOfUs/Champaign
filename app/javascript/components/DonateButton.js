import React from 'react';
import Button from './Button/Button';
import CurrencyAmount from './CurrencyAmount';
import ProcessingThen from './ProcessingThen';
import { FormattedMessage } from 'react-intl';
import './DonateButton.css';

export default props => {
  let buttonString = 'fundraiser.donate';
  if (props.recurring) {
    buttonString = props.weekly
      ? 'fundraiser.donate_weekly'
      : 'fundraiser.donate_monthly';
  }

  return (
    <Button
      className="DonateButton"
      onClick={props.onClick}
      disabled={props.disabled}
    >
      <ProcessingThen processing={props.submitting || false}>
        <span className="fa fa-lock" />
        &nbsp;
        <FormattedMessage
          id={buttonString}
          defaultMessage="Donate {amount}"
          values={{
            amount: (
              <CurrencyAmount
                amount={props.amount || 0}
                currency={props.currency}
              />
            ),
          }}
        />
      </ProcessingThen>
    </Button>
  );
};
