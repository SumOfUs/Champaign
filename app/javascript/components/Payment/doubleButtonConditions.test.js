import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import DonateButton from '../DonateButton';

const disableSubmit = jest.fn();
const onClickHandle = jest.fn();

const currency = '$';
const donationAmount = 1;
const submitting = false;

let showMonthlyButton = (recurringDonor, recurringDefault) => {
  if (recurringDonor) {
    return false;
  } else {
    if (recurringDefault === 'only_one_off') {
      return false;
    }
    return true;
  }
};

let showOneOffButton = (recurringDonor, recurringDefault) => {
  if (recurringDonor) {
    return true;
  } else {
    if (recurringDefault === 'only_one_off') {
      return true;
    }

    if (recurringDefault === 'only_recurring') {
      return false;
    }
    return true;
  }
};

describe('Recurring Donors', () => {
  it(`shows one_off button for recurring donors when recurring_default is any`, () => {
    const recurringDonor = true;
    const recurring_default = 'only_recurring';
    let oneOffWrapper = null;

    if (showOneOffButton(recurringDonor, recurring_default)) {
      oneOffWrapper = shallow(
        <DonateButton
          currency={currency}
          amount={donationAmount || 0}
          submitting={submitting}
          name="one_time"
          recurring={false}
          recurringDonor={recurringDonor}
          disabled={disableSubmit}
          onClick={onClickHandle}
        />
      );
    }

    expect(toJson(oneOffWrapper)).toMatchSnapshot();
  });

  it(`doesn't show recurring button for recurring donors`, () => {
    const recurringDonor = true;
    const recurring_default = 'one_off';
    let recurringWrapper = null;

    if (showMonthlyButton(recurringDonor, recurring_default)) {
      recurringWrapper = shallow(
        <DonateButton
          currency={currency}
          amount={donationAmount || 0}
          submitting={submitting}
          name="one_time"
          recurring={false}
          recurringDonor={recurringDonor}
          disabled={disableSubmit}
          onClick={onClickHandle}
        />
      );
    }

    expect(recurringWrapper).toBeNull();
  });
});

describe('For one_off/non_donor/new_user donors when recurring default is one_off', () => {
  it(`shows one_off button`, () => {
    const recurringDonor = false;
    const recurring_default = 'one_off';
    let oneOffWrapper = null;

    if (showOneOffButton(recurringDonor, recurring_default)) {
      oneOffWrapper = shallow(
        <DonateButton
          currency={currency}
          amount={donationAmount || 0}
          submitting={submitting}
          name="one_time"
          recurring={false}
          recurringDonor={recurringDonor}
          disabled={disableSubmit}
          onClick={onClickHandle}
        />
      );
    }

    expect(toJson(oneOffWrapper)).toMatchSnapshot();
  });

  it(`shows recurring button`, () => {
    const recurringDonor = false;
    const recurring_default = 'one_off';
    let recurringWrapper = null;

    if (showMonthlyButton(recurringDonor, recurring_default)) {
      recurringWrapper = shallow(
        <DonateButton
          currency={currency}
          amount={donationAmount || 0}
          submitting={submitting}
          name="one_time"
          recurring={false}
          recurringDonor={recurringDonor}
          disabled={disableSubmit}
          onClick={onClickHandle}
        />
      );
    }

    expect(toJson(recurringWrapper)).toMatchSnapshot();
  });
});

describe('For one_off/non_donor/new_user donors when recurring default is recurring', () => {
  it(`shows one_off button`, () => {
    const recurringDonor = false;
    const recurring_default = 'recurring';
    let oneOffWrapper = null;

    if (showOneOffButton(recurringDonor, recurring_default)) {
      oneOffWrapper = shallow(
        <DonateButton
          currency={currency}
          amount={donationAmount || 0}
          submitting={submitting}
          name="one_time"
          recurring={false}
          recurringDonor={recurringDonor}
          disabled={disableSubmit}
          onClick={onClickHandle}
        />
      );
    }

    expect(toJson(oneOffWrapper)).toMatchSnapshot();
  });

  it(`shows recurring button`, () => {
    const recurringDonor = false;
    const recurring_default = 'recurring';
    let recurringWrapper = null;

    if (showMonthlyButton(recurringDonor, recurring_default)) {
      recurringWrapper = shallow(
        <DonateButton
          currency={currency}
          amount={donationAmount || 0}
          submitting={submitting}
          name="one_time"
          recurring={false}
          recurringDonor={recurringDonor}
          disabled={disableSubmit}
          onClick={onClickHandle}
        />
      );
    }

    expect(toJson(recurringWrapper)).toMatchSnapshot();
  });
});

describe('For one_off/non_donor/new_user donors when recurring default is only_one_off', () => {
  it(`shows one_off button`, () => {
    const recurringDonor = false;
    const recurring_default = 'only_one_off';
    let oneOffWrapper = null;

    if (showOneOffButton(recurringDonor, recurring_default)) {
      oneOffWrapper = shallow(
        <DonateButton
          currency={currency}
          amount={donationAmount || 0}
          submitting={submitting}
          name="one_time"
          recurring={false}
          recurringDonor={recurringDonor}
          disabled={disableSubmit}
          onClick={onClickHandle}
        />
      );
    }

    expect(toJson(oneOffWrapper)).toMatchSnapshot();
  });

  it(`doesn't show recurring button`, () => {
    const recurringDonor = false;
    const recurring_default = 'only_one_off';
    let recurringWrapper = null;

    if (showMonthlyButton(recurringDonor, recurring_default)) {
      recurringWrapper = shallow(
        <DonateButton
          currency={currency}
          amount={donationAmount || 0}
          submitting={submitting}
          name="one_time"
          recurring={false}
          recurringDonor={recurringDonor}
          disabled={disableSubmit}
          onClick={onClickHandle}
        />
      );
    }

    expect(recurringWrapper).toBeNull();
  });
});

describe('For one_off/non_donor/new_user donors when recurring default is only_recurring', () => {
  it(`doesn't show one_off button`, () => {
    const recurringDonor = false;
    const recurring_default = 'only_recurring';
    let oneOffWrapper = null;

    if (showOneOffButton(recurringDonor, recurring_default)) {
      oneOffWrapper = shallow(
        <DonateButton
          currency={currency}
          amount={donationAmount || 0}
          submitting={submitting}
          name="one_time"
          recurring={false}
          recurringDonor={recurringDonor}
          disabled={disableSubmit}
          onClick={onClickHandle}
        />
      );
    }

    expect(oneOffWrapper).toBeNull();
  });

  it(`shows recurring button`, () => {
    const recurringDonor = false;
    const recurring_default = 'only_recurring';
    let recurringWrapper = null;

    if (showMonthlyButton(recurringDonor, recurring_default)) {
      recurringWrapper = shallow(
        <DonateButton
          currency={currency}
          amount={donationAmount || 0}
          submitting={submitting}
          name="one_time"
          recurring={false}
          recurringDonor={recurringDonor}
          disabled={disableSubmit}
          onClick={onClickHandle}
        />
      );
    }

    expect(toJson(recurringWrapper)).toMatchSnapshot();
  });
});
