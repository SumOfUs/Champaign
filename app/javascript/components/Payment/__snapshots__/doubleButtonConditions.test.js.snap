// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`For one_off/non_donor/new_user donors when recurring default is one_off shows one_off button 1`] = `
<Button
  className="DonateButton"
  disabled={[MockFunction]}
  name="one_time"
  onClick={[MockFunction]}
>
  <_default
    processing={false}
  >
    <span
      className="fa fa-lock"
    />
     
    <FormattedMessage
      defaultMessage="Donate {amount} Just Once"
      id="fundraiser.donation_once"
      values={
        Object {
          "amount": <CurrencyAmount
            amount={1}
            currency="$"
          />,
        }
      }
    />
  </_default>
</Button>
`;

exports[`For one_off/non_donor/new_user donors when recurring default is one_off shows recurring button 1`] = `
<Button
  className="DonateButton"
  disabled={[MockFunction]}
  name="one_time"
  onClick={[MockFunction]}
>
  <_default
    processing={false}
  >
    <span
      className="fa fa-lock"
    />
     
    <FormattedMessage
      defaultMessage="Donate {amount} Just Once"
      id="fundraiser.donation_once"
      values={
        Object {
          "amount": <CurrencyAmount
            amount={1}
            currency="$"
          />,
        }
      }
    />
  </_default>
</Button>
`;

exports[`For one_off/non_donor/new_user donors when recurring default is only_one_off shows one_off button 1`] = `
<Button
  className="DonateButton"
  disabled={[MockFunction]}
  name="one_time"
  onClick={[MockFunction]}
>
  <_default
    processing={false}
  >
    <span
      className="fa fa-lock"
    />
     
    <FormattedMessage
      defaultMessage="Donate {amount} Just Once"
      id="fundraiser.donation_once"
      values={
        Object {
          "amount": <CurrencyAmount
            amount={1}
            currency="$"
          />,
        }
      }
    />
  </_default>
</Button>
`;

exports[`For one_off/non_donor/new_user donors when recurring default is only_recurring shows recurring button 1`] = `
<Button
  className="DonateButton"
  disabled={[MockFunction]}
  name="one_time"
  onClick={[MockFunction]}
>
  <_default
    processing={false}
  >
    <span
      className="fa fa-lock"
    />
     
    <FormattedMessage
      defaultMessage="Donate {amount} Just Once"
      id="fundraiser.donation_once"
      values={
        Object {
          "amount": <CurrencyAmount
            amount={1}
            currency="$"
          />,
        }
      }
    />
  </_default>
</Button>
`;

exports[`For one_off/non_donor/new_user donors when recurring default is recurring shows one_off button 1`] = `
<Button
  className="DonateButton"
  disabled={[MockFunction]}
  name="one_time"
  onClick={[MockFunction]}
>
  <_default
    processing={false}
  >
    <span
      className="fa fa-lock"
    />
     
    <FormattedMessage
      defaultMessage="Donate {amount} Just Once"
      id="fundraiser.donation_once"
      values={
        Object {
          "amount": <CurrencyAmount
            amount={1}
            currency="$"
          />,
        }
      }
    />
  </_default>
</Button>
`;

exports[`For one_off/non_donor/new_user donors when recurring default is recurring shows recurring button 1`] = `
<Button
  className="DonateButton"
  disabled={[MockFunction]}
  name="one_time"
  onClick={[MockFunction]}
>
  <_default
    processing={false}
  >
    <span
      className="fa fa-lock"
    />
     
    <FormattedMessage
      defaultMessage="Donate {amount} Just Once"
      id="fundraiser.donation_once"
      values={
        Object {
          "amount": <CurrencyAmount
            amount={1}
            currency="$"
          />,
        }
      }
    />
  </_default>
</Button>
`;

exports[`Recurring Donors shows one_off button for recurring donors when recurring_default is any 1`] = `
<Button
  className="DonateButton"
  disabled={[MockFunction]}
  name="one_time"
  onClick={[MockFunction]}
>
  <_default
    processing={false}
  >
    <span
      className="fa fa-lock"
    />
     
    <FormattedMessage
      defaultMessage="Donate {amount} Just Once"
      id="fundraiser.donation_once"
      values={
        Object {
          "amount": <CurrencyAmount
            amount={1}
            currency="$"
          />,
        }
      }
    />
  </_default>
</Button>
`;
